controllers/userStatus.js

```diff
@@ -0,0 +1,102 @@
+const userStatusModel = require("../models/userStatus");
+
+/**
+ * Creates a new User Status
+ * @param req {object} - Express Request Object
+ * @param res {object} - Express Response Object
+ */
+
+const addUserStatus = async (req, res) => {
+  try {
+    const { id, userStatusData } = await userStatusModel.addUserStatus({ ...req.body });
+    return res.status(201).json({
+      message: "User Status created successfully",
+      id,
+      ...userStatusData,
+    });
+  } catch (error) {
+    logger.error(`Error while creating new User Status: ${error}`);
+    return res.boom.badImplementation("An internal server error occurred");
+  }
+};
+/**
+ * Deletes a new User Status
+ *
+ * @param req {object} - Express Request Object
+ * @param res {object} - Express Response Object
+ *
+ */
+
+const deleteUserStatus = async (req, res) => {
+  try {
+    const { userId } = req.params;
+    const deletedUserStatus = await userStatusModel.deleteUserStatus(userId);
+    return res.status(200).json({
+      message: "UserStatus Deleted successfully.",
+      ...deletedUserStatus,
+    });
+  } catch (error) {
+    logger.error(`Error while deleting User Status: ${error}`);
+    return res.boom.badImplementation("An internal server error occurred");
+  }
+};
+
+/**
+ * Collects single User Status
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+const getUserStatus = async (req, res) => {
+  try {
+    const { userId } = req.params;
+    const userData = await userStatusModel.getUserStaus(userId);
+    return res.json({ message: "User Status found successfully", ...userData });
+  } catch (err) {
+    // can return 404 error as well
```

Comment:
This provides better information for user as well as easy to debug as well

Reply:
Added a status code 404 for the record not found.